include/hlslib/xilinx/OpenCL.h

```diff
@@ -28,10 +28,50 @@ namespace ocl {
 
 #if HLSLIB_LEGACY_SDX == 0
 constexpr auto kXilinxMemPointer = CL_MEM_EXT_PTR_XILINX;
-constexpr auto kMemoryBank0 = XCL_MEM_DDR_BANK0;
-constexpr auto kMemoryBank1 = XCL_MEM_DDR_BANK1;
-constexpr auto kMemoryBank2 = XCL_MEM_DDR_BANK2;
-constexpr auto kMemoryBank3 = XCL_MEM_DDR_BANK3;
+
+/*
+The Xilinx Alveo U280 FPGA expects different flags for DRAM banks than previous 
+DSAs. Therefore for Xilinx these flags are now initialized and stored as part of the
+Context class
+*/
+class DDRBankFlags {
+public:
+  DDRBankFlags(const std::string &device_name) {
+    if (device_name.find("xilinx_u280") != std::string::npos) {
+      memory_bank_0_ = XCL_MEM_TOPOLOGY | 32;
+      memory_bank_1_ = XCL_MEM_TOPOLOGY | 33;
```

Comment:
You created the `kHBMStorageMagicNumber`, but it seems you are not using it? Maybe you can just get rid of it then?

Reply:
Ok, sure!